src/utils/signaling.js

```diff
@@ -279,18 +278,12 @@ Signaling.Base.prototype.joinCall = function(token, flags, silent, recordingCons
 				resolve()
 				this._joinCallSuccess(token)
 			}.bind(this))
-			.catch(function() {
+			.catch(function(e) {
 				reject(new Error())
-				if (!IS_DESKTOP) {
-					// Server maintenance, lobby kicked in, or room not found.
-					// We first redirect to the conversation again and that
-					// will then show the proper error message to the user.
-					window.location = generateUrl('call/' + token)
-				} else {
-					// TODO: Is it true, reload is equal to generateUrl('call/' + token) here?
-					// Or can we always just reload the page?
-					window.location.reload()
-				}
+				store.commit('connectionFailed', {
+					token,
+					payload: e.response?.data?.ocs?.data?.error,
```

Comment:
should also log the erorr in the console, so it's retrievable by the user?

Reply:
@nickvergessen Just the error code is enough?

Something like this - 

```JS
console.error('Connection failed, reason: ', e.response?.data?.ocs?.data?.error)
```

Reply:
yes